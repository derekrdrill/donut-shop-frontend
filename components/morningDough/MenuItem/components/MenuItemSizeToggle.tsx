import * as React from 'react';
import { Grid, ToggleButtonGroup } from '@mui/material';

import { MenuItemToggleButton } from '../style';

import { FullMenuItem } from '../../Menu/assets/data/FULL_MENU';

type MenuItemSizeToggleProps = {
  menuItemData: FullMenuItem | null | undefined;
  selectedSize: string;
  setSelectedSize: React.Dispatch<React.SetStateAction<string>>;
};

const MenuItemSizeToggle = ({
  menuItemData,
  selectedSize,
  setSelectedSize,
}: MenuItemSizeToggleProps) => {
  const handleChange = (event: React.MouseEvent<HTMLElement>, newSelectedSize: string) => {
    setSelectedSize(newSelectedSize);
  };

  return menuItemData?.category === 'drink' && !menuItemData?.bottled ? (
    <Grid data-testid='size-toggle' item xs={12}>
      <ToggleButtonGroup color='secondary' value={selectedSize} exclusive onChange={handleChange}>
        <MenuItemToggleButton value='sm'>SM</MenuItemToggleButton>
        <MenuItemToggleButton value='md'>MD</MenuItemToggleButton>
        <MenuItemToggleButton value='lg'>LG</MenuItemToggleButton>
        <MenuItemToggleButton value='xl'>XL</MenuItemToggleButton>
      </ToggleButtonGroup>
    </Grid>
  ) : (
    <React.Fragment />
  );
};

export default MenuItemSizeToggle;
