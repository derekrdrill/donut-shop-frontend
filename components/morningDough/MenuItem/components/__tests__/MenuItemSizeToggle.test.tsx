import '@testing-library/jest-dom';
import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';

import MenuItemSizeToggle from '../MenuItemSizeToggle';

describe('MenuItemSizeToggle unit tests', () => {
  it('renders as expected', () => {
    const result = renderer
      .create(
        <MenuItemSizeToggle
          menuItemData={{
            key: 'icedCoffee',
            name: 'Iced Coffee',
            image: 'icedCoffee.png',
            category: 'drink',
            subCategory: 'coldDrinks',
            soda: false,
            bottled: false,
          }}
          selectedSize='md'
          setSelectedSize={jest.fn()}
        />,
      )
      .toJSON();
    expect(result).toMatchSnapshot();
  });

  it('renders MenuItemSizeToggle correctly (size-toggle)', async () => {
    render(
      <MenuItemSizeToggle
        menuItemData={{
          key: 'icedCoffee',
          name: 'Iced Coffee',
          image: 'icedCoffee.png',
          category: 'drink',
          subCategory: 'coldDrinks',
          soda: false,
          bottled: false,
        }}
        selectedSize='md'
        setSelectedSize={jest.fn()}
      />,
    );

    const menuItemSizeToggle = await screen.findAllByTestId('size-toggle');

    expect(menuItemSizeToggle[0]).toHaveClass('MuiGrid-root');
  });
});
