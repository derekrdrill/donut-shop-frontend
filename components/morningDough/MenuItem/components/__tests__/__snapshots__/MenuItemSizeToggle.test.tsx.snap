// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuItemSizeToggle unit tests renders as expected 1`] = `
<div
  className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-13i4rnv-MuiGrid-root"
  data-testid="size-toggle"
>
  <div
    className="MuiToggleButtonGroup-root css-1gjgmky-MuiToggleButtonGroup-root"
    role="group"
  >
    <button
      aria-pressed={false}
      className="MuiButtonBase-root MuiToggleButton-root MuiToggleButton-sizeMedium MuiToggleButton-secondary sc-eDDNvR iWGnQD MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal css-ovja5t-MuiButtonBase-root-MuiToggleButton-root"
      disabled={false}
      onBlur={[Function]}
      onChange={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
      value="sm"
    >
      SM
    </button>
    <button
      aria-pressed={true}
      className="MuiButtonBase-root MuiToggleButton-root Mui-selected MuiToggleButton-sizeMedium MuiToggleButton-secondary sc-eDDNvR iWGnQD MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal css-ovja5t-MuiButtonBase-root-MuiToggleButton-root"
      disabled={false}
      onBlur={[Function]}
      onChange={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
      value="md"
    >
      MD
    </button>
    <button
      aria-pressed={false}
      className="MuiButtonBase-root MuiToggleButton-root MuiToggleButton-sizeMedium MuiToggleButton-secondary sc-eDDNvR iWGnQD MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal css-ovja5t-MuiButtonBase-root-MuiToggleButton-root"
      disabled={false}
      onBlur={[Function]}
      onChange={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
      value="lg"
    >
      LG
    </button>
    <button
      aria-pressed={false}
      className="MuiButtonBase-root MuiToggleButton-root MuiToggleButton-sizeMedium MuiToggleButton-secondary sc-eDDNvR iWGnQD MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal css-ovja5t-MuiButtonBase-root-MuiToggleButton-root"
      disabled={false}
      onBlur={[Function]}
      onChange={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
      value="xl"
    >
      XL
    </button>
  </div>
</div>
`;
